src/elements/content-explorer/__tests__/MetadataSidePanel.test.tsx

```diff
@@ -124,4 +132,152 @@ describe('elements/content-explorer/MetadataSidePanel', () => {
         const submitButton = screen.getByRole('button', { name: 'Save' });
         expect(submitButton).toBeInTheDocument();
     });
+
+    test('switches back to view mode when cancel button is clicked', async () => {
+        renderComponent();
+        const editTemplateButton = screen.getByLabelText('Edit Mock Template');
+        await userEvent.click(editTemplateButton);
+
+        const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+        await userEvent.click(cancelButton);
+
+        // Should be back in view mode
+        expect(screen.getByLabelText('Edit Mock Template')).toBeInTheDocument();
+    });
+
+    test('calls onUpdate when form is submitted for single item', async () => {
+        const mockUpdateMetadata = jest.fn().mockResolvedValue(undefined);
+        renderComponent({ onUpdate: mockUpdateMetadata });
+
+        const editTemplateButton = screen.getByLabelText('Edit Mock Template');
+        await userEvent.click(editTemplateButton);
+
+        const submitButton = screen.getByRole('button', { name: 'Save' });
+        await userEvent.click(submitButton);
+
+        expect(mockUpdateMetadata).toHaveBeenCalledWith(
+            [mockCollection.items[0]],
+            expect.any(Array),
+            expect.any(Array),
+            expect.any(Array),
+            expect.any(Function),
+            expect.any(Function),
+        );
+    });
+
+    test('calls onUpdate when multiple items are selected', async () => {
+        const mockUpdateMetadata = jest.fn().mockResolvedValue(undefined);
+
+        renderComponent({
+            selectedItemIds: new Set(['1', '2']),
+            onUpdate: mockUpdateMetadata,
+        });
+
+        const editTemplateButton = screen.getByLabelText('Edit Mock Template');
+        await userEvent.click(editTemplateButton);
+
+        const submitButton = screen.getByRole('button', { name: 'Save' });
+        await userEvent.click(submitButton);
+
+        await waitFor(() => {
+            expect(mockUpdateMetadata).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    test('displays success notification when metadata update succeeds', async () => {
+        const mockUpdateMetadata = jest.fn().mockImplementation((_, __, ___, ____, successCallback) => {
+            successCallback();
+            return Promise.resolve();
+        });
+        const mockRefreshCollection = jest.fn();
+
+        renderComponent({
+            onUpdate: mockUpdateMetadata,
+            refreshCollection: mockRefreshCollection,
+        });
+
+        const editTemplateButton = screen.getByLabelText('Edit Mock Template');
+        await userEvent.click(editTemplateButton);
+
+        const submitButton = screen.getByRole('button', { name: 'Save' });
+        await userEvent.click(submitButton);
+
+        await waitFor(() => {
```

Comment:
try to limit only the necessary assertion in the waitFor:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback